src/components/_video/VideoTileBase/VideoTileCover.tsx

```diff
@@ -67,144 +67,154 @@ type VideoTileCoverProps = {
 export const MIN_VIDEO_TILE_WIDTH = 250
 const SMALL_SIZE_WIDTH = 300
 
-export const VideoTileCover: React.FC<VideoTileCoverProps> = ({
-  hasAssetUploadFailed,
-  videoHref,
-  openInNewBrowserTab,
-  setTileSize,
-  tileSize,
-  onRemoveButtonClick,
-  onClick,
-  isLoading,
-  uploadStatus,
-  thumbnailUrl,
-  isLoadingThumbnail,
-  isDraft,
-  isUnlisted,
-  publisherMode,
-  duration,
-  onPullupClick,
-  removeButton,
-  thumbnailAlt,
-}) => {
-  const isUploading = uploadStatus && uploadStatus.lastStatus !== 'completed'
-  const { ref: imgRef } = useResizeObserver<HTMLImageElement>({
-    onResize: (size) => {
-      const { width: videoTileWidth } = size
-      if (videoTileWidth) {
-        if (tileSize !== 'small' && videoTileWidth < SMALL_SIZE_WIDTH) {
-          setTileSize('small')
+export const VideoTileCover: React.FC<VideoTileCoverProps> = React.memo(
+  ({
+    hasAssetUploadFailed,
+    videoHref,
+    openInNewBrowserTab,
+    setTileSize,
+    tileSize,
+    onRemoveButtonClick,
+    onClick,
+    isLoading,
+    uploadStatus,
+    thumbnailUrl,
+    isLoadingThumbnail,
+    isDraft,
+    isUnlisted,
+    publisherMode,
+    duration,
+    onPullupClick,
+    removeButton,
+    thumbnailAlt,
+  }) => {
+    const isUploading = useMemo(() => uploadStatus && uploadStatus.lastStatus !== 'completed', [uploadStatus])
```

Comment:
Do we really need `useMemo` here? Seems like a very simple calculation and it also outputs `boolean` which is primitive so shouldn't trigger props change for children?

Reply:
I made this change on the basis of `Why Did You Render` output, it showed that this variable was causing unnecessary rerenders.